extensions/pkg/webhook/utils_test.go

```diff
@@ -15,143 +15,963 @@
+			var pvcs []corev1.PersistentVolumeClaim
+
+			BeforeEach(func() {
+				pvcs = []corev1.PersistentVolumeClaim{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "pvc1",
+						},
+						Spec: corev1.PersistentVolumeClaimSpec{
+							Resources: corev1.ResourceRequirements{
+								Requests: map[corev1.ResourceName]resource.Quantity{
+									"storage": resource.MustParse("10Gi"),
+								},
+							},
+						},
 					},
-				},
-			}
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "pvc2",
+						},
+						Spec: corev1.PersistentVolumeClaimSpec{
+							Resources: corev1.ResourceRequirements{
+								Requests: map[corev1.ResourceName]resource.Quantity{
+									"storage": resource.MustParse("20Gi"),
+								},
+							},
+						},
+					},
+				}
+			})
 
-			actual := webhook.EnsureFileWithPath(files, newFile)
-			Expect(actual).To(Equal([]extensionsv1alpha1.File{
-				{
-					Path: "/foo.txt",
+			It("should add a new PersistentVolumeClaim if not present", func() {
+				newPVC := corev1.PersistentVolumeClaim{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "pvc3",
+					},
+					Spec: corev1.PersistentVolumeClaimSpec{
+						Resources: corev1.ResourceRequirements{
+							Requests: map[corev1.ResourceName]resource.Quantity{
+								"storage": resource.MustParse("30Gi"),
+							},
+						},
+					},
+				}
+				result := webhook.EnsurePVCWithName(pvcs, newPVC)
+				Expect(result).To(Equal(append(pvcs, newPVC)))
+			})
+
+			It("should replace the existing PersistentVolumeClaim if it's not identical", func() {
+				existingPVC := corev1.PersistentVolumeClaim{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "pvc1",
+					},
+					Spec: corev1.PersistentVolumeClaimSpec{
+						Resources: corev1.ResourceRequirements{
+							Requests: map[corev1.ResourceName]resource.Quantity{
+								"storage": resource.MustParse("30Gi"),
+							},
+						},
+					},
+				}
+				result := webhook.EnsurePVCWithName(pvcs, existingPVC)
+				Expect(result).To(Equal([]corev1.PersistentVolumeClaim{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "pvc1",
+						},
+						Spec: corev1.PersistentVolumeClaimSpec{
+							Resources: corev1.ResourceRequirements{
+								Requests: map[corev1.ResourceName]resource.Quantity{
+									"storage": resource.MustParse("30Gi"),
+								},
+							},
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "pvc2",
+						},
+						Spec: corev1.PersistentVolumeClaimSpec{
+							Resources: corev1.ResourceRequirements{
+								Requests: map[corev1.ResourceName]resource.Quantity{
+									"storage": resource.MustParse("20Gi"),
+								},
+							},
+						},
+					},
+				}))
+			})
+
+			It("should do nothing to the existing PersistentVolumeClaim if it's identical", func() {
+				identicalPVC := pvcs[0]
+				result := webhook.EnsurePVCWithName(pvcs, identicalPVC)
+				Expect(result).To(Equal(pvcs))
+			})
+		})
+
+		Describe("#EnsureFileWithPath", func() {
```

Comment:
There are now 2 `Describe`s for "#EnsureFileWithPath". Because I also added tests for this func in https://github.com/gardener/gardener/pull/9121. Let's drop 1 of the `Describe`s for  "#EnsureFileWithPath".